### DISCUSSION TITLE
C-MOVE via dcm4che in MirthConnect ?

### DISCUSSION THREAD

**Discussion by p-warman** [nextgenhealthcare/connect#4840] [Q&A]
Hi All,

I'm looking to create a channel where the receipt of a specific HL7 message would trigger a C-MOVE request from an external PACS system back to a separate MirthConnect channel. 

Online I couldn't find any information on doing C-MOVE within MirthConnect, but I understand that one can add custom libraries to resources. I see that there is a Java setup of dcm4che that has C-MOVE implementation (movescu): https://github.com/dcm4che/dcm4che

I'm a bit lost though on how to include this repo as a MirthConnect Resource and then call the movescu command in the destination script (or does the destination need to be a TCP sender?), and would appreciate any advice or suggestions. If there is an alternative approach please let me know. 

Thanks!

**Comment by jonbartels** [comment]
I am not a DICOM expert so I can only provide some pointers or ideas and not a solution.

There are some older forum threads discussing C-MOVE at: [https://forums.mirthproject.io/search?q=C-MOVE&searchJSON=%7B%22keywords%22%3A%22C-MOVE%22%7D](https://forums.mirthproject.io/search?q=C-MOVE&searchJSON=%7B%22keywords%22%3A%22C-MOVE%22%7D)

There is also a large DICOM project at https://github.com/nextgenhealthcare/connect/projects and also several open issues and discussions on GitHub. Issues discussing updating dcm4che may show how to deploy the updated dcm4che libraries as a workaround and that would then let you use the `movescu` implementation.
